src/core/qgssettings.h

```diff
@@ -224,33 +225,126 @@ class CORE_EXPORT QgsSettings : public QObject
      * Return the setting value for a setting based on an enum.
      * This forces the output to be a valid and existing entry of the enum.
      * Hence if the setting value is incorrect, the given default value is returned.
-     * If \a flag is true, the value is checked for a flag definition.
+     * This tries first with setting as a string (as the enum) and then as an integer value.
      * \note The enum needs to be declared with Q_ENUM, and flags with Q_FLAG (not Q_FLAGS).
+     * \see setEnumValue
+     * \see flagValue
      */
     template <class T>
     T enumValue( const QString &key, const T &defaultValue,
-                 const Section section = NoSection, bool flag = false ) const
+                 const Section section = NoSection ) const
     {
+      QMetaEnum metaEnum = QMetaEnum::fromType<T>();
+      if ( !metaEnum.isValid() )
+      {
+        QgsDebugMsg( "Invalid metaenum. Enum probably misses Q_ENUM or Q_FLAG declaration." );
```

Comment:
Could this be a `Q_ASSERT` to prevent devs from using this the wrong way?

Reply:
or is the usage widespread throughout the code and we need this for legacy fallback?